### PR TITLE
[dynamo] Fix comparison between SymNodeVariable and ListVariable

### LINKED ISSUE #109504: [dynamo] torch._dynamo.exc.Unsupported: comparison SymNodeVariable() <built-in function is_> ListVariable()
### 🐛 Describe the bug

Self explanatory.

Occurred when implementing self-referentiality check in torch.tensor ref

```
            if cur_item is obj:
                raise TypeError("new(): self-referential lists are incompatible")
```

### Versions

main

cc @anijain2305 @chauhang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @msaroufim @wconstab @bdhirsh @Xia-Weiwen @aakhundov

### PR DESCRIPTION
**Pull request by vdesai2014**
Fixes #109504

Shuffled around an if statement to cause mis-matching types with 'is' operator to return False instead of triggering unimplemented exception

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_dynamo/variables/builtin.py [hunk 1]
```diff
@@ -1365,6 +1365,11 @@ def _comparison(self, tx, left, right):
 
         def _unimplemented():
             unimplemented(f"comparison {typestr(left)} {op} {typestr(right)}")
+
```
Comment (zou3519) [inline torch/_dynamo/variables/builtin.py:1368]: Can you add a test?

**Comment by ezyang** [comment]
looks good, test plz

File: torch/_dynamo/variables/builtin.py [hunk 2]
```diff
@@ -1365,6 +1365,11 @@ def _comparison(self, tx, left, right):
 
         def _unimplemented():
             unimplemented(f"comparison {typestr(left)} {op} {typestr(right)}")
+        
+        if op.__name__ == "is_":
+            # If the two objects are of different type, we can safely return False
+            if type(left) is not type(right):
+                return ConstantVariable.create(False)
```
Comment (jon-chuang) [inline torch/_dynamo/variables/builtin.py:1372]: This is incorrect. `is_` doesn't just mean type. It actually means equality of objects.

```
>>> MyClass(1) is MyClass(1)
False
>>> type(MyClass(1)) == type(MyClass(1))
True
```

Comment (jon-chuang) [reply torch/_dynamo/variables/builtin.py:1372]: In particular, this is not good enough. SymNodeVar is never of type ListVariable. Lol.
